lib/rack/request.rb

```diff
@@ -638,9 +707,33 @@ def reject_trusted_ip_addresses(ip_addresses)
         ip_addresses.reject { |ip| trusted_proxy?(ip) }
       end
 
+      FORWARDED_SCHEME_HEADERS = {
+        proto: HTTP_X_FORWARDED_PROTO,
+        scheme: HTTP_X_FORWARDED_SCHEME
+      }.freeze
+      private_constant :FORWARDED_SCHEME_HEADERS
       def forwarded_scheme
-        allowed_scheme(get_header(HTTP_X_FORWARDED_SCHEME)) ||
-        allowed_scheme(extract_proto_header(get_header(HTTP_X_FORWARDED_PROTO)))
+        forwarded_priority.each do |type|
+          case type
```

Comment:
I feel using a case statement here is a little bit cumbersome. Like, we know in advance what code should be executed. 

Thinking longer term, does the symbol list in `forwarded_priority` make it hard to evolve this implementation? If we wanted to get rid of the case statement and make the internal implementation more streamlined.

Does it matter? My main thought is performance. I guess probably not. If it was a big issue, maybe later we can construct a lambda based on the expected behaviour and just cache it?

I guess the alternative would be, instead of an array (which is flexible, and could even contain the same entry more than once even thought it doesn't make sense).

Maybe an alternative would be `self.send(:"forwareded_scheme_#{type}")`???

Reply:
I'm not attached to the particular approach used.  If we want to change to use `send` instead of `case`, that seems fine with me.  The important part is that the solution implemented allows the user to choose which header to respect in which order, and allows configuring to explicitly ignore either header or both headers.

Reply:
Do you have the energy to make that change and do you prefer it? I'm happy to defer to your judgement here.

Reply:
I'm fine keeping the `case`.  We can optimize using `send` later in a backwards compatible manner if it turns out to be a bottleneck.